### PR TITLE
Package Index Generation

### PR DESCRIPTION
**Pull request by scbedd**
Includes:

* backwards-compatible `generate_versioned_index.py` that generates the package index
* updates to the current `generate_doc.py` pipeline for prepping for a single run of sphinx to generate all documentation. 
* injecting additional js/css/template extensions to allow direct upload of CI-generated individual package sites

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: eng/tox/tox.ini [hunk 1]
```diff
@@ -34,6 +34,8 @@ platform = linux: linux
            macos: darwin
            windows: win32
 passenv = *
+setenv =
```
Comment (scbedd) [inline eng/tox/tox.ini]: @lmazuel I'll check this in tomorrow. This is the sphinx-options addition I stated I'd add.

File: doc/sphinx/_static/js/get_options.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+      nameElements = Array.from(xmlDoc.getElementsByTagName('Name'))
+      options = []
+
+      for (var i in nameElements){
+        options.push(nameElements[i].textContent.split('/')[3])
+      }
+
+      for (var i in options){
+        $(selector).append('<li><a target="new" href="' + getPackageUrl('python', packageName, options[i]) + '">' + options[i] + '</a></li>')
```
Comment (JimSuplizio) [inline doc/sphinx/_static/js/get_options.js]: Should use the SELECTED_LANGUAGE here

File: doc/sphinx/_static/js/get_options.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+function hideSelectors(selectors){
+  selectors.forEach(function(item, index){
+    $(item).hide()
+  })
+}
+
+function populateOptions(optionSelector, otherSelectors){
+  if(currentPackage()){
+    var versionRequestUrl = "https://azuresdkdocs.blob.core.windows.net/$web?restype=container&comp=list&prefix=python/" + currentPackage() + "/versions/"
```
Comment (JimSuplizio) [inline doc/sphinx/_static/js/get_options.js]: Should use the SELECTED_LANGUAGE here instead of the hard coded 'python' string

**Comment by scbedd** [comment]
Resolves #7789

**Pull request merged**